lr-cli/cortx_setup/commands/server/config.py

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from ..command import Command
+from cortx.utils.conf_store import Conf
+from cortx_setup.commands.common_utils import get_machine_id
+from cortx_setup.config import CONFSTORE_CLUSTER_FILE
+from provisioner.salt import local_minion_id
+from provisioner.commands import PillarSet
+
+
+class ServerConfig(Command):
+
+    _args = {
+        'name': {
+            'type': str,
+            'default': None,
+            'optional': True,
+            'help': 'Set name of the server'
+        },
+        'type': {
+            'type': str,
+            'default': None,
+            'optional': True,
+            'choices': ['VM', 'HW'],
+            'help': 'Server type e.g {VM|HW}'
+        }
+    }
+
+
+    def run(self, **kwargs):
+        """
+        Server configuration command execution method
+
+        Command:
+        `cortx_setup server config --name <name> --type {VM|HW}}`
+        """
+
+        node_id = local_minion_id()
+        machine_id = get_machine_id(node_id)
+
+        conf_pillar_map = {
+            'name': (
```

Comment:
@mazinamdar this looks shabby to capture both pillar and confstore format without clear demarcation.
Can this be improvised as:
```
{
  'name': (
    'pillar_key': f'cluster/{node_id}/hostname',
    'confstore_key': f'server_node>{machine_id}>name'
  ),
  'type': (
    'pillar_key': f'cluster/{node_id}/node_type',
    'confstore_key': f'server_node>{machine_id}>type'
  )
}
```

Reply:
@ypise yes we can use nested dictionary or rather we can have some convention defined as in index 0 for pillar_key and index 1 for confstore key for such logic in our codebase - this will reduce the redundant witing of the "pillar_key" and "confstore_key" for each argument

Reply:
>index 0 for pillar_key and index 1 for confstore key
In the long run that turns out to be a pain area. We have faced that with Networking definition in config.ini in R1.

Reply:
Comment addressed!
created 2 different dictionaries pillar_key_map and conf_key_map for better demarcation of pillar and confstore keys